tests/scenario/modules/assets/asset-group.test.ts

```diff
@@ -0,0 +1,330 @@
+import _ from "lodash";
+import {
+  assetGroupTestId,
+  assetGroupTestBody,
+  assetGroupTestParentId,
+  assetGroupTestParentBody,
+  assetGroupTestChildrenId,
+  assetGroupTestChildrenBody,
+} from "../../../fixtures/assetsGroups";
+
+// Lib
+import {
+  ApiGroupCreateRequest,
+  ApiGroupCreateResult,
+  ApiGroupDeleteRequest,
+  ApiGroupReadRequest,
+  ApiGroupSearchRequest,
+  ApiGroupSearchResult,
+  ApiGroupUpdateRequest,
+  ApiGroupAddAssetsRequest,
+  AssetsGroupsBodyRequest,
+} from "../../../../lib/modules/asset/types/AssetGroupsAPI";
+import { AssetsGroupsBody } from "../../../../lib/modules/asset/types/AssetGroupContent";
+import { setupHooks } from "../../../helpers";
+
+jest.setTimeout(10000);
+
+describe("AssetsGroupsController", () => {
+  const sdk = setupHooks();
+  const baseRequest = {
+    controller: "device-manager/assetsGroup",
+    engineId: "engine-ayse",
+  } as const;
+
+  // Tests data
+  const assetGroupRootId = "root-group";
+  const assetGroupChildrenId = "children-group";
+
+  const assetGroupRootBodyRequest: AssetsGroupsBodyRequest = {
+    name: "root group",
+    children: [assetGroupChildrenId],
+  };
+  const assetGroupRootBody: AssetsGroupsBody = {
+    ...assetGroupRootBodyRequest,
+    parent: null,
+  };
+
+  const assetGroupChildrenBody: AssetsGroupsBody = {
+    name: "children group",
+    children: [],
+    parent: assetGroupRootId,
+  };
+
+  it("can create a group", async () => {
+    const missingIdQuery: Omit<ApiGroupCreateRequest, "_id"> = {
+      ...baseRequest,
+      action: "create",
+      body: assetGroupRootBodyRequest,
+    };
+    await expect(sdk.query(missingIdQuery)).rejects.toThrow(
+      /^Missing argument "_id".$/
+    );
+
+    const missingBodyQuery: Omit<ApiGroupCreateRequest, "body"> = {
+      ...baseRequest,
+      action: "create",
+      _id: assetGroupRootId,
+    };
+    await expect(sdk.query(missingBodyQuery)).rejects.toThrow(
+      /^The request must specify a body.$/
+    );
+
+    const badParentIdQuery: ApiGroupCreateRequest = {
+      ...baseRequest,
+      action: "create",
+      _id: "parent-not-exist",
+      body: {
+        name: "Parent not exist",
+        parent: "not-exist",
+      },
+    };
+    await expect(sdk.query(badParentIdQuery)).rejects.toThrow(
+      /^The parent group "not-exist" does not exist$/
+    );
+
+    const { result: assetGroupRoot } = await sdk.query<
+      ApiGroupCreateRequest,
+      ApiGroupCreateResult
+    >({
+      ...baseRequest,
+      action: "create",
+      _id: assetGroupRootId,
+      body: _.omit(assetGroupRootBodyRequest, "children"),
+    });
```

Comment:
Same, it's very opinionated, but I think you should not do magic tricks in your query (`...baseRequest` or use lodash's `.omit()`).
Your queries should appear "raw", you should duplicate them if necessary, because it's easier to maintain and more readable if you want to look at the test to understand the purpose of the feature.

Reply:
ok it's changed